src/components/Star/util.ts

```diff
@@ -0,0 +1,28 @@
+import { createContext, useContext } from 'react'
+
+ /* By default, React requires an initial context. This requires
+ * code duplication, and becomes annoying as state grows. We can
+ * eliminate the duplication, get type safe Context, and even get a 
+ * nice error message by using a proxy object
+ */
+export function createContextWithDefault<C>(): [React.Context<C>, () => C] {
```

Comment:
I didn't use this, but this is a little Util I made myself a long time ago that I use on basically all React projects nowadays. It lets you spin up a ReactContext object that is type safe without having to create a default blank object, which I find to be super annoying, especially with large amounts of state. If you like it, we can lift it to the appropriate location.